chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -21,6 +35,88 @@
 namespace {
 
 constexpr SkColor kBraveBrandColor = SkColorSetRGB(0xff, 0x76, 0x54);
+constexpr SkColor kBravePrimaryColor = SkColorSetRGB(32, 74, 227);
+SkColor AddOpacity(SkColor color, float opacity) {
+  DCHECK(opacity >= 0 && opacity <= 1);
+  auto current_alpha = SkColorGetA(color);
+  return SkColorSetA(color, current_alpha * opacity);
+}
+
+using Kind = views::MdTextButton::Kind;
+using ColorScheme = ui::NativeTheme::PreferredColorScheme;
+using ButtonState = views::Button::ButtonState;
+
+struct ButtonStyle {
+  absl::optional<SkColor> background_color;
+  absl::optional<SkColor> border_color;
+  SkColor text_color;
+};
+
+struct MdTextButtonStyleKey {
+  Kind kind;
+  ColorScheme color_scheme;
+  ButtonState state;
+
+  bool operator<(const MdTextButtonStyleKey& other) const {
+    return std::tie(kind, color_scheme, state) <
+           std::tie(other.kind, other.color_scheme, other.state);
+  }
+};
+
+constexpr float kDisabledOpacity = 0.5f;
+constexpr float kLoadingOpacity = 0.8f;
+
+const base::flat_map<MdTextButtonStyleKey, ButtonStyle> g_button_themes = {
```

Comment:
pls put this into a function with `base::NoDestructor` instance inside.

Reply:
Done :smile: